File: showdown-go/entity/exchange_hands.go
package entity

import players "showdown-go/entity/player"

type ExchangeHands struct {
	Duration     int
	Trader       *players.Player
	Counterparty *players.Player
}

func NewExchangeHands(trader *players.Player, counterparty *players.AbstractPlayer) *ExchangeHands {
	return &ExchangeHands{
		Trader:       trader,
		Counterparty: counterparty,
		Duration:     3,
	}
}

func (e *ExchangeHands) Countdown() {
	e.Duration--
}

func (e *ExchangeHands) GetTrader() *player.Player {
	return e.Trader
}

func (e *ExchangeHands) GetCounterparty() *player.Player {
	return e.Counterparty
}

func (e *ExchangeHands) GetDuration() int {
	return e.Duration
}

File: Showdown/src/ExchangeHands.php
<?php

namespace Showdown;

use Showdown\Players\Player;

class ExchangeHands
{
    private int $duration;

    private Player $trader;

    private Player $counterparty;

    public function __construct(Player $trader, Player $counterparty)
    {
        $this->trader = $trader;
        $this->counterparty = $counterparty;
        $this->duration = 3;
    }

    public function countdown(): void
    {
        $this->duration--;
    }

    public function getTrader(): Player
    {
        return $this->trader;
    }

    public function getCounterparty(): Player
    {
        return $this->counterparty;
    }

    public function getDuration(): int
    {
        return $this->duration;
    }
}
